Xamarin.Essentials/Connectivity/Connectivity.shared.cs

```diff
@@ -38,27 +31,14 @@ public static partial class Connectivity
             }
         }
 
-        static void SetCurrent()
-        {
-            currentAccess = NetworkAccess;
-            currentProfiles = new List<ConnectionProfile>(Profiles);
-        }
-
         static void OnConnectivityChanged(NetworkAccess access, IEnumerable<ConnectionProfile> profiles)
             => OnConnectivityChanged(new ConnectivityChangedEventArgs(access, profiles));
 
         static void OnConnectivityChanged()
             => OnConnectivityChanged(NetworkAccess, Profiles);
 
         static void OnConnectivityChanged(ConnectivityChangedEventArgs e)
-        {
-            if (currentAccess != e.NetworkAccess ||
-                !currentProfiles.SequenceEqual(e.Profiles))
-            {
-                SetCurrent();
-                Platform.BeginInvokeOnMainThread(() => ConnectivityChanagedInternal?.Invoke(e));
-            }
-        }
+            => Platform.BeginInvokeOnMainThread(() => ConnectivityChanagedInternal?.Invoke(e));
```

Comment:
Any reason not to compare? I fear of this as things can fire even when things haven't changed.

Reply:
Does the OS report a change, but nothing has actually changed?

Reply:
it is highly possible. Sometimes things trigger back and forth. This was a big complaint with my connectivity plugin that it didn't do enough checks. We can test though I guess..... Maybe put a counter on the sample page?

Reply:
I saw this:
https://github.com/jamesmontemagno/BatteryPlugin/issues/1
Let me add the filter back.